### ISSUE TITLE
(SDL) ppsspp does not always resize window when toggling fullscreen

### ISSUE THREAD

**Issue by orbea** [hrydgard/ppsspp#11991]
### What happens?

When the SDL fronted is started as fullscreen and the user attempts to toggle fullscreen from the menu ppsspp will change successfully become windowed, but the window size will still be fullscreen.

To reproduce:

1. Start the SDL frontend with when fullscreen is already enabled.
2. Toggle fullscreen in the menu.
3. ppsspp will not be fullscreen, but the window will still cover the entire screen making it effectively fullscreen.

When started as a windowed program the original window size will (Probably?) not be the same size as when fullscreen and toggling fullscreen works out just fine.

This is followup to issue https://github.com/hrydgard/ppsspp/issues/11625 where fullscreen has been problematic with the SDL frontend for a long time.

### What *should* happen?

When ppsspp is toggled from fullscreen it should make sure the window size and the fullscreen size are not the same. If they are the same it should automatically resize ppsspp to a smaller window. The default size for when starting ppsspp as a windowed program would suffice.

### What hardware, operating system, and PPSSPP version?  On desktop, GPU matters for graphical issues.

OS: `Slackware64-current`
ppsspp: https://github.com/hrydgard/ppsspp/commit/d15a6cf6bff3bea6adbebeaf2ee6dd973d381b19
SDL: `2.0.9`
GPU: `AMD RX Vega 56` (amdgpu + radeonsi)

**Comment by unknownbrackets** [comment]
This issue is being closed as a duplicate of #11974.

This is an issue tracker - there's a [forum if you want to discuss things](http://forums.ppsspp.org/).  You wouldn't add another "bread" item to a grocery list every time you thought about it, right?  We do the same here - we already have "bread" (this issue) on our list.

By keeping the list clean, we can also make sure all the information a developer needs to fix the issue is at their finger tips.  When there are multiple issues, information gets lost and it becomes hard to fix things, so they are fixed more slowly.  Duplicates won't make fixes appear faster.

And remember - PPSSPP is open source.  Try to find *new and useful information*, and add it to the original issue.  If you can't wait any longer, muster up your courage and try fixing it yourself.  PPSSPP gets better the more people contribute.  You could be that amazing person who solves this problem for everyone.

-[Unknown]

**Comment by orbea** [comment]
@unknownbrackets This is not a duplicate, please reopen. The issues are different.

**Comment by orbea** [comment]
@hrydgard Can you please reopen this issue? The linked issue is not remotely similar outside of both involving the fullscreen toggle. Its really discouraging when it seems that created issues are not even being read...

**Comment by hrydgard** [comment]
Though, I have to say that they do sound very similar...

**Comment by orbea** [comment]
Thanks! Unlike that issue when the windows resize it seems to work as expected, I have a problem with how it will resize from fullscreen resolution to a windowed fullscreen resolution.

From my observations when toggling from fullscreen to windowed ppsspp will resize to the original window size it started with even if that is the same as fullscreen.

Additionally this is not specific to vulkan or any other video backend.

**Comment by orbea** [comment]
@hrydgard Do these images help you understand?

* First ppsspp starts normally as a windowed program.
![windowed](https://i.imgur.com/h7kKgpF.png)

* I can toggle it fullscreen/windowed normally.
![fullscreen](https://i.imgur.com/hlEiAi6.png)

What should happen is that it resizes the window to be like the first picture instead of giving me a fullscreen window.

* If I close ppsspp while fullscreen and start it again it will still be fullscreen as expected. I can't resize the window since its fullscreen.

* However if go to toggle fullscreen to windowed, its still fullscreen. Note the checkbox is not marked.
![windowed2](https://i.imgur.com/bPGCr70.png)

* It really is windowed though, see how I can resize the window.
![windowed3](https://i.imgur.com/l3OaqmY.png)

**Comment by vnctdj** [comment]
Ah, this last post definitely helps!
Then I feel like this issue should be renamed to something like "Toggling off fullscreen should restore default window size", no?

**Comment by orbea** [comment]
Sure that works, done.

**Comment by orbea** [comment]
I would like to also add that when the window is started with fullscreen disabled and the player toggles the fullscreen it mostly works at first, but after toggling it multiple times in a row the game resolution does not match the window size and you only see a portion of the content. This appears to be vulkan only.

**Comment by hrydgard** [comment]
That latter part with the portion of the content should be fixed in the latest builds. It's not?

**Comment by orbea** [comment]
Yes you are right, I rebuilt the master (https://github.com/hrydgard/ppsspp/commit/a0f8ad41d1a61bf7d488e3f3b945374396f24cb0) and that last part is now fixed.
